### PR TITLE
Integrate NestJS

### PR DESCRIPTION
**Pull request by Jad31**
Add a NestJS application example to Moon.

This NestJS example is integrated into the repository and working with all the existing tasks listed in the [README.md](https://github.com/moonrepo/examples/blob/master/README.md) at the root.

### CODE REVIEW

**Comment by Jad31** [comment]
Hi @milesj, could you take a look and figure out why there are errors with ESLint? I seem to have introduced these errors while using NestJS, but I can't determine how and why.

File: apps/nestjs-app/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    project: 'tsconfig.json',
+    tsconfigRootDir: __dirname,
+    sourceType: 'module',
+  },
+  plugins: ['@typescript-eslint/eslint-plugin'],
+  extends: [
+    'plugin:@typescript-eslint/recommended',
+    'plugin:prettier/recommended',
+  ],
+  root: true,
```
Comment (milesj) [inline apps/nestjs-app/.eslintrc.js:13]: This shouldn't overwrite the root. You can remove every setting here except for `rules`.

File: apps/nestjs-app/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (milesj) [inline apps/nestjs-app/.prettierrc:1]: remove this file

File: apps/nestjs-app/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "reflect-metadata": "^0.1.13",
+    "rxjs": "^7.2.0"
+  },
+  "devDependencies": {
+    "@nestjs/cli": "^9.0.0",
+    "@nestjs/schematics": "^9.0.0",
+    "@nestjs/testing": "^9.0.0",
+    "@types/express": "^4.17.13",
+    "@types/supertest": "^2.0.11",
+    "@typescript-eslint/eslint-plugin": "^5.0.0",
```
Comment (milesj) [inline apps/nestjs-app/package.json:21]: remove these 4 eslint deps, theyll be inherited from root config

File: apps/nestjs-app/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "compilerOptions": {
```
Comment (milesj) [inline apps/nestjs-app/tsconfig.json:2]: ideally this extends the root `tsconfig.options.json`, and we can remove any option here that overlaps

File: apps/nestjs-app/package.json (hunk 3 above)
Comment (Jad31) [reply apps/nestjs-app/package.json:21]: I need `@nestjs/testing` and `supertest\ for `e2e` tests in the file `apps/nestjs-app/test/app.e2e-spec.ts`.
Should I create an 'e2e' task in the `moon.yml` of the project to run the following command: `jest --config ./test/jest-e2e.json`? This is the command that Nest provides when creating a new project in the scripts section of the `package.json`.

Comment (milesj) [reply apps/nestjs-app/package.json:21]: Yeah that makes sense.

File: apps/nestjs-app/tsconfig.json (hunk 4 above)
Comment (milesj) [reply apps/nestjs-app/tsconfig.json:2]: Looks like nest is using an older version of TS.

Comment (Jad31) [reply apps/nestjs-app/tsconfig.json:2]: You're correct. The command I used, `npx -p @nestjs/cli nest new nestjs-app`, actually sets up the application with NestJS version 9, which is using an older version of TypeScript (4.7.3). We should use the command `npx nest new nestjs-app` instead that utilize NestJS version 10 which is associated with the same version of TypeScript than Moon (5.1.3 actually). Thank you for pointing out the version of TS. I'll update the documentation accordingly!

**Comment by milesj** [comment]
Looks good just need to fix the build.

**Comment by milesj** [comment]
Awesome, thanks!

**Pull request merged**